### ISSUE TITLE
Interfaces for people with dyslexia

### ISSUE THREAD

**Issue by ajjackson** [accessibleresearchsoftware/accessibleresearchsoftware#3]
There are various ideas and recommendations out there regarding accessibility for users with dyslexia, some of it contradictory. Some expert input and/or review of academic literature would be helpful. This paper has a few citations and recommends sans-serif fonts but I'm not sure how well-regarded it is https://doi.org/10.1145/2513383.2513447 [[PDF](http://dyslexiahelp.umich.edu/sites/default/files/good_fonts_for_dyslexia_study.pdf)]

The obvious issue here is font selection. For a command-line tool the software engineer will not usually force a particular font on the user, but in GUI design and plotting there is some scope to use this knowledge. Page layout and spacing concerns may also come into play, especially if you have a fancy HTML/CSS interface. Web user-interface people think about this stuff a lot, so we should look out for resources aimed at that crowd, e.g. [UI blog posts](http://blog.usabilla.com/how-to-design-for-dyslexia).

Some people may prefer to use a screen-reader. We should aim to learn more about designing for screen-readers.

**Comment by ajjackson** [comment]
This resource was suggested by Tania during group discussion: http://www.bdadyslexia.org.uk/common/ckeditor/filemanager/userfiles/About_Us/policies/Dyslexia_Style_Guide.pdf

A few of the pointers seem quite relevant for text-based command-interfaces, especially:
> * AVOID TEXT IN BLOCK CAPITALS: this is much harder to read.
> ...
> * Boxes and borders can be used for effective emphasis.
> * Lines should not be too long: 60 to 70 characters.

And also some tips for screen-reader compatibility

**Comment by trallard** [comment]
yes that resource provides some good insight @ajjackson I am unsure if this will fall here @yochannah so feel free to move this elsewhere

People with some degrees of blindness will use a  [screen-reader](https://en.wikipedia.org/wiki/Screen_reader) as noted above, meaning they rely on the text where we might get visual cues from icons, images, and other richly rendered content.

In order to provide context to these cues, it is a good practice to create a `screen-reader-text` class (also found as `sr-only`. Where anything you place in such a class will be hidden for screens while still providing the text for the screen readers.

Simple example here:

```css
.screen-reader-text {
    clip: rect(1px, 1px, 1px, 1px);
    position: absolute !important;
    height: 1px;
    width: 1px;
    overflow: hidden;
}
```
Use this instead of `display: none` or `visibility: hidden`

**Comment by ajjackson** [comment]
There seems to be a fair amount that can be done for websites. Does this mean the web-apps are a better platform than native GUIs for developing accessible applications? If the user has a screen-reader set up for their default web browser then it should be pretty happy to read a web app. Style sheets can be used for font and colour-scheme tweaks.

**Comment by yochannah** [comment]
I think that the web has a *lot* of work put into native controls being accessible and usable for screen readers and potentially other disabilities - one problem I've encountered is that web developers don't always use them properly - for example, if a dropdown menu doesn't look pleasing / match with the theme of the website, they might write their own from scratch (without implementing good practice usability fixes), and suddenly the screen reader doesn't know what to do with it.

I guess I'd answer your question with a "yes" but cautiously - depending on if good practice is followed or not.

**Comment by trallard** [comment]
I do a lot of web-app based work so I might be a bit biased and that might also be the reason why I am aware of these resources. I agree, however, that the stylesheets and also some JavaScripts make it *easy* to tweak the apps. 

I am sure many of these recommendations could be easily adapted for native GUIs although this might mean additional digging around. 

Either way,  as @yochannah mentioned you should try and adhere to good practices
